src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -65,6 +71,37 @@ export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement
         this.namespace = getNamespace(DATA_IMPORT_BATCH_OBJECT.objectApiName);
     }
 
+    connectedCallback() {
+        registerListener('geBatchGiftEntryTableChangeEvent', this.retrieveBatchTotals, this);
+    }
+
+    disconnectedCallback() {
+        unregisterListener('geBatchGiftEntryTableChangeEvent', this.retrieveBatchTotals, this);
+    }
+
+    async retrieveBatchTotals() {
+        this.batchTotals = await BatchTotals(this.batchId);
+
+        if(this.batchTotals.hasPaymentsWithExpiredAuthorizations &&
+            !this._hasPaymentsWithExpiredAuthorizations) {
+            const detail = {
+                modalProperties: {
+                    componentName: 'geModalPrompt',
+                    showCloseButton: false
+                },
+                componentProperties: {
+                    'severity': 'warning',
+                    'title': this.CUSTOM_LABELS.gePaymentAuthExpiredHeader,
+                    'message': this.CUSTOM_LABELS.gePaymentAuthExpiredWarningText,
+                    'buttonText': this.CUSTOM_LABELS.commonOkay
+                },
+            };
+            this.dispatchEvent(new CustomEvent('togglemodal', { detail }));
+            this._hasPaymentsWithExpiredAuthorizations = true;
```

Comment:
Is this to guard against the modal popping up multiple times?

Reply:
Yes - and definitely welcome feedback for this approach. Without this in place, you would see the modal pop up every time you added a gift and there was an expired auth present, since this function is called on the geBatchGiftEntryTableChangeEvent. I believe that the intent is that you only see this notification once-per-page-load, but this feels a little bit clunky and maybe there's a better way to accomplish the goal?

Reply:
Gotcha, thank you for clarifying! I think this is fine. My only suggestion would be to update the name to be specific to the modal display. Maybe along the lines of `_hasShownExpiredAuthorizationWarning` or something?